langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbTestFixture.java

```diff
@@ -0,0 +1,126 @@
+        this.client = client;
+    }
+
+    MongoDbTestFixture initialize() {
+        return this.initialize(b -> b);
+    }
+
+    MongoDbTestFixture initialize(Function<MongoDbEmbeddingStore.Builder, MongoDbEmbeddingStore.Builder> initializer) {
+        this.embeddingStore = initializer.apply(getDefaultMongoDbEmbeddingStoreBuilder()).build();
+        return this;
+    }
+
+    static void assertDoContainerTests() {
+        Assumptions.assumeTrue(getenv("MONGODB_DISABLE_CONTAINER_TESTS") == null);
+    }
+
+    EmbeddingStore<TextSegment> getEmbeddingStore() {
+        return embeddingStore;
+    }
+
+    private static MongoDBAtlasLocalContainer getContainer() {
+        if (mongodb == null) {
+            mongodb = new MongoDBAtlasLocalContainer("mongodb/mongodb-atlas-local:7.0.9");
+            mongodb.start();
+        }
+        return mongodb;
+    }
+
+    static MongoClient createClientFromContainer() {
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(getContainer().getConnectionString()));
+        return createClientFromBuilder(builder);
+    }
+
+    static MongoClient createClientFromEnv() {
+        boolean hasParts = HOST != null && USERNAME != null && PASSWORD != null;
+        Assumptions.assumeTrue(hasParts || CONNECTION_STRING != null,
+                "Skipping test: host/username/password, or connection string must be provided via env vars.");
+
+        String connectionString = CONNECTION_STRING != null
+                ? CONNECTION_STRING
+                : format("mongodb+srv://%s:%s@%s/?retryWrites=true&w=majority", USERNAME, PASSWORD, HOST);
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(connectionString));
+        return createClientFromBuilder(builder);
+    }
+
+    private static MongoClient createClientFromBuilder(MongoClientSettings.Builder builder) {
+        MongoClientSettings mongoClientSettings = builder
+                .uuidRepresentation(UuidRepresentation.STANDARD)
+                .serverApi(ServerApi.builder().version(ServerApiVersion.V1).build())
+                .build();
+        return MongoClients.create(mongoClientSettings);
+    }
+
+    private MongoDbEmbeddingStore.Builder getDefaultMongoDbEmbeddingStoreBuilder() {
+        IndexMapping indexMapping = IndexMapping.builder()
+                .dimension(EMBEDDING_MODEL.dimension())
+                .metadataFieldNames(Sets.newHashSet("test-key"))
+                .build();
+        return MongoDbEmbeddingStore.builder()
+                .fromClient(client)
+                .databaseName(DATABASE_NAME)
+                .collectionName(collectionName)
+                .indexName("test_index")
+                .indexMapping(indexMapping)
+                .createIndex(true);
+    }
+
+    void afterTests() {
```

Comment:
collectionName might be passed as a parameter. If it is unique, dropping collection is not required

Reply:
I am not sure why it would need to be passed in, since it is being tracked as a field, or what "unique" means. But, (unless, of course, there is some problem with this PR that you could identify) I suggest such enhancements might be performed in a future PR, if they are necessary.